src/test/java/org/apache/commons/lang3/concurrent/LazyInitializerSupplierTest.java

```diff
@@ -44,4 +50,56 @@ public void testIsInitialized() throws ConcurrentException {
         assertTrue(initializer.isInitialized());
     }
 
+    @Test
+    public void testSupplierThrowsCheckedException() {
+        final LazyInitializer<Object> initializer = LazyInitializer.builder().setInitializer(
+                () -> methodThatThrowsException(ExceptionToThrow.IOException)).get();
+        assertThrows(ConcurrentException.class, () -> initializer.get());
+    }
+
+    @Test
+    public void testSupplierThrowsRuntimeException() {
+        final LazyInitializer<Object> initializer = LazyInitializer.builder().setInitializer(
+                () -> methodThatThrowsException(ExceptionToThrow.NullPointerException)).get();
+        assertThrows(NullPointerException.class, () -> initializer.get());
+    }
+
+    /*
+     * This method tests that if LazyInitializer.initialize catches a ConcurrentException it will rethrow it without wrapping it
+     */
+    @Test
+    public void testSupplierThrowsConcurrentException() {
+        final ConcurrentException concurrentException = new ConcurrentException();
+        final LazyInitializer<Object> initializer = LazyInitializer.builder().setInitializer(
+                () -> {
+                    if ("test".equals("test")) {
+                        throw concurrentException;
+                    }
+                    return new Object();
+                } ).get();
+        try {
+        	initializer.get();
+        	fail();
+        } catch (ConcurrentException e) {
+        	assertEquals(concurrentException, e);
+        }
+    }
+
+    private enum ExceptionToThrow {
+    	IOException,
+    	SQLException,
+    	NullPointerException
+    }
+
+    private static String methodThatThrowsException(ExceptionToThrow input) throws IOException, SQLException {
```

Comment:
Use `final` where you can.

Reply:
Checkstyle complains about a redundant final modifier if I do.

Reply:
Do you mean when you change `ExceptionToThrow input` to `final ExceptionToThrow input`?

Reply:
That would explain it, I thought you meant final before the method. I was quite confused for a bit.